migrations/800_groups__add.sql

```diff
@@ -0,0 +1,29 @@
+-- A Work in Progress to discuss shared schema for group storage in PL
+
+CREATE TABLE IF NOT EXISTS group_configs (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    assessment_id BIGINT REFERENCES assessments(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    type TEXT, -- what will these be? This might belong in groups rather than groups_config
+    name TEXT, -- or maybe description is accurate here? Maybe as a base for groups created from it?
+    minimum INT,
+    maximum INT,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE IF NOT EXISTS groups (
+    id BIGSERIAL PRIMARY KEY,
+    course_instance_id BIGINT NOT NULL REFERENCES course_instances(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    name TEXT,      -- visible name of the group
+    group_config_id BIGINT REFERENCES group_configs(id) ON DELETE CASCADE ON UPDATE CASCADE,
+    date timestamp with time zone DEFAULT CURRENT_TIMESTAMP,
+    deleted_at timestamp with time zone
+);
```

Comment:
Should we include a UNIQUE (course_instance_id, name) here, to catch conflicts?

Reply:
I'm in favor of adding constraints wherever possible.

Reply:
I added the uniqueness constraints based on `course_instance_id, name, group_config_id`. It made sense to me that if a `group_config` was used, the name should be unique inside of it but could be reused across two different configurations (i.e. Assessment 1 and Assessment 2 can both have a "Group 1"

Two partial constraints were added because postgresql doesn't see NULL as unique and I wanted to catch the case where no group_config was applied still had uniqueness based on the name and course_instance_id. We might be able to simplify this if we require a linked group_config in groups instead of making it optional.